.ci/build

```diff
@@ -13,9 +13,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -e
+set -eo pipefail
 
-source .ci/common.sh
+# For all steps, concourse will set the following environment variables:
+# SOURCE_PATH - path to component repository root directory.
+if [[ -z "${SOURCE_PATH}" ]]; then
```

Comment:
setting of `SOURCE_PATH` is done in both `.ci/build` and `.ci/test` - would it make sense to also move this to `.ci/common.sh`?

Reply:
That was the initial approach in https://github.com/gardener/etcd-druid/pull/517. But that did not work because the `.ci/build` and `.ci/test` scripts were unable to source `common.sh` unless they knew the repo root (absolute path) and sourced other scripts built using the absolute path. So in this PR I was forced to move the logic of determining `SOURCE_PATH` back into the individual scripts as the only way of fixing the [error we saw after merging that PR to master](https://concourse.ci.gardener.cloud/teams/gardener/pipelines/etcd-druid-master/jobs/master-head-update-job/builds/5).